uds/diagnostic_configuration/ecu_configuration.py

```
@@ -0,0 +1,327 @@
+    @subfunction_restrictions.setter
+    def subfunction_restrictions(self, value: Mapping[SidAlias, Mapping[int, RequiredStatesAlias]]) -> None:
+        """
+        Set ECU restrictions for SubFunctions handling.
+
+        :param value: Value to set.
+
+        :raise TypeError: Value is not Mapping type.
+        :raise ValueError: Keys in the Mapping are not SID values only.
+        """
+        if not isinstance(value, Mapping):
+            raise TypeError(f"Provided value is not a Mapping. Actual type: {type(value)}.")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)